### PR TITLE
Adding container image for running examples

### PR DESCRIPTION
**Pull request by bovem**
[Short description explaining the high-level reason for the pull request]

## Changes

- Added a dockerfile for the image that could be used for running examples.
- Each example folder has a python virtual environment named `hamilton` that should be activated first before running the example.

## Testing

- Build the container image
```bash
docker build --tag hamilton-example .
```

- Creating a container
```bash
docker run -it --rm --name hamilton-example hamilton-example
```

- Running the `hello_world` example inside the container
```bash
cd hamilton/example/hello_world
source hamilton-env/bin/activate
python my_script.py
```

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Python - local testing

- [ ] python 3.6
- [ ] python 3.7
- [x] python 3.9 (Docker Environment)

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -21,3 +21,23 @@ folders.
 Under `model_examples` you'll find a how you could apply Hamilton to model your ML workflow.
 Check it out to get a sense for how Hamilton could make your ML pipelines reusable/general
 components...
+
+## Running examples through docker image
+- Build the container image
+```bash
+docker build --tag hamilton-example .
```
Comment (skrawcz) [inline examples/README.md:28]: what directory should people be in?

File: examples/README.md [hunk 2]
```diff
@@ -21,3 +21,23 @@ folders.
...
+```
+
+Alternatively, you can pull the container image from here: ...
+
+- Creating a container
+```bash
+docker run -it --rm --name hamilton-example hamilton-example
+```
+
+- Running the `hello_world` example inside the container
```
Comment (skrawcz) [inline examples/README.md:38]: can you add the instruction for starting the container please?

File: examples/make_python_virtualenv.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Get a list of all the folders containing "requirements.txt" file
```
Comment (skrawcz) [inline examples/make_python_virtualenv.sh:1]: Can you add a comment at the top as to the purpose of this file please? 
E.g. "Use this to create individual virtual environments for each example - it assume virtualenv has been installed"?

File: examples/README.md [hunk 4]
```diff
@@ -21,3 +21,23 @@ folders.
 Under `model_examples` you'll find a how you could apply Hamilton to model your ML workflow.
 Check it out to get a sense for how Hamilton could make your ML pipelines reusable/general
 components...
+
+## Running examples through docker image
```
Comment (skrawcz) [inline examples/README.md:25]: Add more to say how this works please.

File: examples/README.md (hunk 1 above)
Comment (skrawcz) [reply examples/README.md:28]: also mention how long this might take.

File: examples/README.md (hunk 4 above)
Comment (bovem) [reply examples/README.md:25]: Added documentation for docker image

File: examples/make_python_virtualenv.sh (hunk 3 above)
Comment (bovem) [reply examples/make_python_virtualenv.sh:6]: Added description of the script at the top. Along with its usage

File: examples/README.md (hunk 1 above)
Comment (bovem) [reply examples/README.md]: Added build time and directory details.

File: examples/README.md (hunk 2 above)
Comment (bovem) [reply examples/README.md]: Added note to skip (to step 3 container creation) if image is already present.

**Comment by elijahbenizzy** [comment]
Saw @skrawcz took a look. Two more things:
1. Make sure to install./run pre-commit checks on everything (the pre-commit is failing). See https://github.com/stitchfix/hamilton/blob/5111491e135df5c07257ca45f06e45cd2477c2e3/developer_setup.md#L16. You may want to reset to the merging branch then pack it all into a single commit and let that run the pre-commit for you.
2. Add yourself to the contributor list!

**Comment by skrawcz** [comment]
@bovem let me know when this PR is back up and ready -- it's on my list to pull down and verify things myself. Thanks :)

**Comment by bovem** [comment]
Hi @skrawcz I forced pushed some commits by mistake and it closed this this PR. I have created this PR: https://github.com/stitchfix/hamilton/pull/209 from a new branch with `pre-commit` checks.
I am sorry if it caused any inconvenience.

**Pull request closed without merging**